src/analysis/Environments/Environments.ts

```diff
@@ -238,29 +206,37 @@ export function PauseButton({ cloudEnvironment, currentUser, pauseComputeAndRefr
     : null;
 }
 
-interface DecoratedResourceAttributes {
-  workspace: WorkspaceInfo;
-  unsupportedWorkspace: boolean;
+export interface UseWorkspacesResult {
```

Comment:
```suggestion
export type UseWorkspacesStateResult {
```

See below comment for name.

And nit about mixing `interface` and `type`.

Reply:
I will argue that while they are generally interchangeable... interfaces edge out the two overall if there is a choice.

https://www.typescriptlang.org/play#example/types-vs-interfaces

Reply:
I think we've had this discussion before. I prefer always using `type` because:
- It must be used for some cases: declaring a union type, etc. Thus, if we must use `type` some of the time, might as well use it all of the time and reduce the number of concepts present in the code.
- `type` behaves more intuitively. `interface` has the surprising behavior of allowing extending an interface by declaring it multiple times.

Reply:
but how about this from that link I listed above:

// That said, we recommend you use interfaces over type
// aliases. Specifically, because you will get better error
// messages. If you hover over the following errors, you can
// see how TypeScript can provide terser and more focused
// messages when working with interfaces like Chicken.

Reply:
Today I learned. Ok, I'm sold on interfaces.

Reply:
... and there was much rejoicing!

Great debate though.  Never be shy to question stuff like this.